project/Build.scala

```diff
@@ -707,125 +706,7 @@ object Build {
-
-          // Files must be Jars, non-files must be dirs
-          val (jars, dirs) = cp.filter(_.exists).partition(_.isFile)
-          val irFiles = dirs.flatMap(dir => (dir ** "*.sjsir").get)
-
-          def seqOfStringsToJSArrayCode(strings: Seq[String]): String =
-            strings.map(s => "\"" + escapeJS(s) + "\"").mkString("[", ", ", "]")
-
-          val irPaths = {
-            val absolutePaths = (jars ++ irFiles).map(_.getAbsolutePath)
-            seqOfStringsToJSArrayCode(absolutePaths)
-          }
-
-          val mainMethods = {
-            /* Ideally we would read `scalaJSModuleInitializers in (testSuite, Test)`,
-             * but we cannot convert the ModuleInitializers to strings to be
-             * passed to the QuickLinker (because ModuleInitializer is a
-             * write-only data structure). So we have some duplication.
-             */
-            val unescapedMainMethods = List(
-                "org.scalajs.testsuite.compiler.ModuleInitializerInNoConfiguration.main",
-                "org.scalajs.testsuite.compiler.ModuleInitializerInTestConfiguration.main2",
-                "org.scalajs.testsuite.compiler.ModuleInitializerInTestConfiguration.main1",
-                "org.scalajs.testsuite.compiler.ModuleInitializerInTestConfiguration.mainArgs1()",
-                "org.scalajs.testsuite.compiler.ModuleInitializerInTestConfiguration.mainArgs2(foo,bar)"
-            )
-            seqOfStringsToJSArrayCode(unescapedMainMethods)
-          }
-
-          val scalaJSEnvForTestSuite = {
-            s"""
-            {"javaSystemProperties": {
-              "scalajs.scalaVersion": "${scalaVersion.value}",
-              "scalajs.testsuite.testtag": "testtag.value",
-              "scalajs.nodejs": "true",
-              "scalajs.typedarray": "true",
-              "scalajs.fastopt-stage": "true",
-              "scalajs.modulekind-nomodule": "true"
-            }}
-            """
-          }
-
-          val code = {
-            s"""
-            var toolsTestModule = require("${escapeJS(toolsTestModulePath)}");
-            var linker = toolsTestModule.scalajs.QuickLinker;
-            var lib = linker.linkTestSuiteNode($irPaths, $mainMethods);
-
-            var __ScalaJSEnv = $scalaJSEnvForTestSuite;
-            eval("(function() { 'use strict'; " +
-              lib + ";" +
-              "scalajs.ConsoleTestRunner.runTests();" +
-            "}).call(this);");
-            """
-          }
-
-          val launcher = new MemVirtualJSFile("Generated launcher file")
-            .withContent(code)
-
-          val config = RunConfig()
-            .withLogger(sbtLogger2ToolsLogger(streams.value.log))
-
-          val input = Input.ScriptsToLoad((executionFiles :+ launcher).toList)
-
-          val run = jsEnv.value.start(input, config)
-          Await.result(run.future, Duration.Inf)
-        }
-      }
+      scalaJSLinkerConfig in Test ~= (_.withModuleKind(ModuleKind.CommonJSModule))
```

Comment:
I was wondering why it was necessary, since the Node.js facades should be dead-code-eliminated away. But in fact it's used from [`Platform.loadJar`](https://github.com/scala-js/scala-js/blob/a04778779f96ebcb6608ecd38bf8bc2830e23307/linker/js/src/test/scala/org/scalajs/linker/testutils/Platform.scala#L7) through `LinkerTest`.

So it's more than acceptable; it makes sense that it be necessary.